zomes/deepkey_csr/src/app_binding.rs

```diff
@@ -42,13 +44,34 @@ pub fn query_app_binding_by_index(index: u32) -> ExternResult<(ActionHash, AppBi
 }
 
 #[hdk_extern]
-pub fn query_app_binding_by_id(installed_app_id: String) -> ExternResult<(ActionHash, AppBinding)> {
+pub fn query_app_binding_by_action(addr: ActionHash) -> ExternResult<AppBinding> {
+    Ok(
+        query_app_bindings(())?
+            .into_iter()
+            .find( |(app_binding_addr, _)| *app_binding_addr == addr  )
+            .ok_or(guest_error!(format!("No AppBinding with action hash: {}", addr )))?.1
+    )
+}
+
+#[hdk_extern]
+pub fn query_app_bindings_by_installed_app_id(installed_app_id: String) -> ExternResult<Vec<(ActionHash, AppBinding)>> {
+    Ok(
+        query_app_bindings(())?
+            .into_iter()
+            .filter( |(_, app_binding)| app_binding.installed_app_id == installed_app_id  )
+            .collect()
+    )
+}
+
+#[hdk_extern]
+pub fn query_app_binding_by_key(key_bytes: ByteArray<32>) -> ExternResult<(ActionHash, AppBinding)> {
+    let (ka_action_addr, _) = crate::key_anchor::get_first_key_anchor_for_key( key_bytes )?;
```

Comment:
Is this key just "any agent key that's ever been bound to this app"?

Reply:
Yes.  It should always trace back to the first key anchor and derive the app binding